### PR TITLE
Backend events

### PR DESCRIPTION
**Pull request by Oporto**
Get and place events through the events servlet

### CODE REVIEW

File: src/main/java/com/google/codeu/data/Location.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    private double lat;
+    private double lng;
+    private String locationName;
+    private String country;
+    private String state;
+    private String city;
+    private int zipcode;
+    private List<String> adressLines;
+    
+    public Location(double lat, double lng, String name, String country, String state, String city, int zipcode, List<String> adressLines){
```
Comment (pnvt23) [inline src/main/java/com/google/codeu/data/Location.java]: Can you add some documentation for the constructor?

Usually for this many parameters, it's good idea to use Builder pattern.

File: src/main/java/com/google/codeu/data/Location.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
+package com.google.codeu.data;
+
+import java.util.List;
+
+public class Location {
+
```
Comment (pnvt23) [inline src/main/java/com/google/codeu/data/Location.java:6]: you should have at most 1 empty line

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 3]
```diff
@@ -26,71 +26,225 @@
...
-    List<Message> messages = new ArrayList<>();
-    PreparedQuery results = datastore.prepare(query);
+    /**
+    * Gets messages received by a specific user (recipient).
+    *
+    * @return a list of messages received by a user, or empty list if user has
+    *         never received a message. List is sorted by time descending.
+    */
+    public List<Message> getMessages(String recipient) {
+        Query query =
```
Comment (pnvt23) [inline src/main/java/com/google/codeu/data/Datastore.java]: Can you fix indentation for this file and the rest ?

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 4]
```diff
@@ -26,71 +26,225 @@
...
+                String externalLink =  (String) entity.getProperty("externalLink");
+                PublicType publicType =  (PublicType) entity.getProperty("publicType");
+                int ownerId =  (int) entity.getProperty("ownerId");
+                List<ThreadComment> thread =  getThread(eventId);
+                long timeStamp =  (long) entity.getProperty("timestamp");
+            
+                Event event = new Event(eventId, timeStamp, speaker, organization, eventDate, location, amenities, externalLink, publicType, ownerId);
+                event.copyThread(thread);
+                events.add(event);
+            } catch (Exception e) {
```
Comment (pnvt23) [inline src/main/java/com/google/codeu/data/Datastore.java]: Usually it's good practice to only catch a specific set of Exception, and throw the rest

If there's a need to catch all exception, please add a comment here.

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 5]
```diff
@@ -26,71 +26,225 @@
...
+                amenities.add(text);
+            } catch (Exception e) {
+                System.err.println("Error reading message.");
+                System.err.println(entity.toString());
+                e.printStackTrace();
+            }
+        }
+        
+        return amenities;
+
```
Comment (pnvt23) [inline src/main/java/com/google/codeu/data/Datastore.java]: Remove empty lines

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 6]
```diff
@@ -26,71 +26,225 @@
...
+        
+        return amenities;
+        
+        
+    }
+    
+    public List<ThreadComment> getThread(UUID eventId){
+        List<ThreadComment> thread = new ArrayList<>();
+        
+        Query query = new Query("ThreadComment").setFilter(new Query.FilterPredicate("eventId", FilterOperator.EQUAL, eventId.toString()))
```
Comment (pnvt23) [inline src/main/java/com/google/codeu/data/Datastore.java]: can you check the character limit of this line

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 7]
```diff
@@ -26,71 +26,225 @@
...
 
-        messages.add(message);
-      } catch (Exception e) {
-        System.err.println("Error reading message.");
-        System.err.println(entity.toString());
-        e.printStackTrace();
-      }
+    }
+  
+    public List<UserMarker> getMarkers() {
```
Comment (pnvt23) [inline src/main/java/com/google/codeu/data/Datastore.java]: I would encourage to document public methods

File: src/main/java/com/google/codeu/data/Datastore.java (hunk 3 above)
Comment (Oporto) [reply src/main/java/com/google/codeu/data/Datastore.java]: I fixed indentation and somehow it things that I replaced a big chunk of code, but not edits to the text were made

File: src/main/java/com/google/codeu/data/Datastore.java (hunk 7 above)
Comment (Oporto) [reply src/main/java/com/google/codeu/data/Datastore.java]: I also updated some methods that should be private

File: src/main/java/com/google/codeu/data/Datastore.java (hunk 4 above)
Comment (Oporto) [reply src/main/java/com/google/codeu/data/Datastore.java]: Added this line: //An exception can occur here for multiple reasons (Type casting error, any property not existing, key error, etc...)

**Review by Oporto — COMMENTED** [review]
Addressed all comments

**Review by pnvt23 — COMMENTED** [review]
I still  see the indentation issue for java file. Can you take another look?

It should be +2sp for new block of code, and +4sp if you need to break into multiple lines.

**Comment by Oporto** [comment]
I thought it would be 4sp no matter what. Should be fixed now

**Review by pnvt23 — APPROVED** [review]

**Pull request merged**